### PR TITLE
Merge flags field into common record

### PR DESCRIPTION
**Pull request by rgrinberg**
This removes a bunch of extraneous conversions and shares decoding code. I will do something similar for the C flags.

### CODE REVIEW

File: src/env_node.ml [hunk 1]
```diff
@@ -119,45 +116,30 @@ let rec c_flags t ~profile ~expander ~default_context_flags =
...
       | Some cfg ->
-        let expander = Expander.set_dir expander ~dir:t.dir in
-        let eval = Expander.expand_and_eval_set expander in
-        let f = cfg.c_flags in
-        if  Ordered_set_lang.Unexpanded.has_special_forms f then
-          eval f ~standard:default
-        else
-          eval f ~standard:(Build.return [])
+        C.Kind.Dict.mapi cfg.c_flags ~f:(fun ~kind f -> lazy (
+          let expander = Expander.set_dir expander ~dir:t.dir in
```
Comment (diml) [inline src/env_node.ml]: `expander` and `eval` are the same in both case, they can go outside of the closure. BTW, given the previous discussions we had about lazy values, I feel a bit wary of this `lazy`. In practice, it buys us very little, so I'm in favour of making things eager here.

**Comment by rgrinberg** [comment]
Yeah agreed. I will get rid of the extra laziness as it complicates things quite a bit.

File: src/env_node.ml (hunk 1 above)
Comment (rgrinberg) [reply src/env_node.ml]: I got rid of it. Agreed that it buys very little.

**Comment by rgrinberg** [comment]
I'm merging this now as I need it for other work and I've addressed Jeremie's main concern.

**Pull request merged**